triedb/pathdb/journal.go

```diff
@@ -193,7 +195,7 @@ func newJournalReader(file string, db ethdb.Database) (JournalReader, error) {
 // loadJournal tries to parse the layer journal from the disk.
 func (db *Database) loadJournal(diskRoot common.Hash) (layer, error) {
 	start := time.Now()
-	reader, err := newJournalReader(db.config.JournalFilePath, db.diskdb)
+	reader, err := newJournalReader(db.config.JournalFilePath, db.diskdb, db.DetermineJournalTypeByStorage())
```

Comment:
If the journalFile flag is specified, but the file doesn't exist, won't an error be reported?

I think it should return an error, rather than use kv type by default as the PR do.